test/paths.ts

```diff
@@ -186,3 +186,27 @@ function testDepth2() {
     Assert<Assignable<P, "nested.tuple.2.good">>,
   ];
 }
+
+/**
+ * Configuration allows wildcard accessors for array indices.
+ */
+function testAnyArrayIndexAccessor() {
+  type With = Paths<ComplexNestedRequired, { anyArrayIndexAccessor: "*" }>;
+  type Without = Paths<ComplexNestedRequired>;
+
+  type cases_with_any_accessor = [
+    Assert<Has<With, "nested.array.*">>,
+    Assert<Has<With, "nested.array.*.bar">>,
+    Assert<Has<With, `nested.array.${number}`>>,
+    Assert<Has<With, `nested.array.${number}.bar`>>,
+    Assert<Has<With, "nested.tuple.0">>,
+    Assert<Has<With, "nested.tuple.2.good">>,
+  ];
```

Comment:
I felt like adding this might be appropriate as I'm not sure that `nested.tuple.*` makes semantic sense. 

```typescript
    // @ts-expect-error: tuples should not be wildcard-able
    Assert<Has<With, "nested.tuple.*">>,
    // @ts-expect-error: tuples should not be wildcard-able
    Assert<Has<With, "nested.tuple.*.good">>,
```

However I could not get it working, trying several avenues of differentiating tuples from arrays - I must be misunderstanding something. This doesn't block my application but if you agree and would rather conservative exposure, I'm going to need some help with it.

Reply:
Give `` `${number}` `` a try as I recommended above, it should help.

Tuples have specific keys: they are numeric literal types (e.g. 0, 1, 2). A conditional type with `DefaultPathsOptions["anyArrayIndexAccessor"]` and `` `${Key}` `` in `type AnyArrayIndexAccessorOrKey` should be sufficient to differentiate tuples and arrays

Reply:
Ah of course, this makes more sense.

I was trying a combination along the lines of this and it would either always swallow all arrays or all tuples:
```typescript
IsTuple<Type> extends Type 
  ? A 
  : Type extends any[]
  ? B
  : C
```

But I agree, the condition ends up much simpler when the `Key` is already inferred as `number`.